sensors/test_deps.go

```diff
@@ -27,47 +28,103 @@ var (
+	// FinishedReplayLidar is a lidar whose NextPointCloud function returns an end of dataset error.
+	FinishedReplayLidar TestSensor = "finished_replay_lidar"
+
+	// GoodIMU is an IMU that works as expected and returns linear acceleration and angular velocity values.
+	GoodIMU TestSensor = "good_imu"
+	// IMUWithErroringFunctions is an IMU whose functions return errors.
+	IMUWithErroringFunctions TestSensor = "imu_with_erroring_functions"
+	// IMUWithInvalidProperties is an IMU whose properties are invalid.
+	IMUWithInvalidProperties TestSensor = "imu_with_invalid_properties"
+	// NoIMU is an IMU that represents that no IMU is set up or added.
+	NoIMU TestSensor = ""
+
+	// ReplayIMU is an IMU that works as expected and returns linear acceleration and angular velocity values.
+	ReplayIMU TestSensor = "replay_imu"
+	// InvalidReplayIMU is an IMU whose meta timestamp is invalid.
+	InvalidReplayIMU TestSensor = "invalid_replay_imu"
+	// FinishedReplayIMU is an IMU whose LinearAcceleration and AngularVelocity functions return an end of
+	// dataset error.
+	FinishedReplayIMU TestSensor = "finished_replay_imu"
+
+	// GoodOdometer is an odometer that works as expected and returns position and orientation values.
+	GoodOdometer TestSensor = "good_odometer"
+
+	// MovementSensorNotIMUNotOdometer is a movement sensor that does neither support an IMU nor an odometer.
+	MovementSensorNotIMUNotOdometer TestSensor = "movement_sensor_not_imu_not_odometer"
+
+	// MovementSensorBothIMUAndOdometer is a movement sensor that dsupports both an IMU nor an odometer.
+	MovementSensorBothIMUAndOdometer TestSensor = "movement_sensor_imu_and_odometer"
+	// GibberishMovementSensor is a movement sensor that can't be found in the dependencies.
+	GibberishMovementSensor TestSensor = "gibberish_movement_sensor"
+	// MovementSensorWithErroringPropertiesFunc is a movement sensor whose Properties function returns an error.
+	MovementSensorWithErroringPropertiesFunc TestSensor = "movement_sensor_with_erroring_properties_function"
+)
+
+var (
+	testLidars = map[TestSensor]func() *inject.Camera{
+		GoodLidar:                  getGoodLidar,
+		WarmingUpLidar:             getWarmingUpLidar,
+		LidarWithErroringFunctions: getLidarWithErroringFunctions,
+		LidarWithInvalidProperties: getLidarWithInvalidProperties,
+		ReplayLidar:                func() *inject.Camera { return getReplayLidar(TestTimestamp) },
+		InvalidReplayLidar:         func() *inject.Camera { return getReplayLidar(BadTime) },
+		FinishedReplayLidar:        getFinishedReplayLidar,
+	}
+
+	testMovementSensors = map[TestSensor]func() *inject.MovementSensor{
+		GoodIMU:                                  getGoodIMU,
+		IMUWithErroringFunctions:                 getIMUWithErroringFunctions,
+		IMUWithInvalidProperties:                 getIMUWithInvalidProperties,
+		ReplayIMU:                                func() *inject.MovementSensor { return getReplayIMU(TestTimestamp) },
```

Comment:
[minor] I think we want to get rid of imu specific test sensors for the most part. Replay IMU -> ReplayMovementSensor... and maybe one testSensor for only odom, and only imu

Reply:
This will be handled in the next PR :)